compiler/src/dotty/tools/dotc/transform/IsInstanceOfChecker.scala

```diff
@@ -88,19 +88,33 @@ object Checkable {
       }
     }
 
+    def stripTypeParam(implicit ctx: Context) = new ApproximatingTypeMap {
+      def apply(tp: Type): Type = tp match {
+        case tp: TypeRef if tp.underlying.isInstanceOf[TypeBounds] =>
```

Comment:
Probably equivalent to `case tp: TypeRef if tp.symbol.isTypeParam`. Maybe clearer?

Reply:
`tp.symbol.isTypeParam` doesn't cover the case of type members, so I think it's better to keep the original version here.